### PR TITLE
Feature/hugo malhar

### PR DESCRIPTION
**Pull request by HugoTse**
Wikipedia Semantic Search with Cohere Embedding Archives 

* Used the Amazon Bedrock InvokeModel API because the Converse API does not support embeddings models.
* Highlighted that Cohere embed is the first embedding model that natively supports int8/byte and binary embeddings.
* Explained how binary encoded embeddings result in a 32x reduction in required memory compared to float32 embeddings.
* Used the November 2023 Wikipedia dataset that Cohere published on Hugging Face because the embeddings dimensionality fits the cohere.embed-multilingual-v3 model available on Bedrock (1024 vs 768).
* Used a query about Alan Turing instead of the founder of Wikipedia because the new dataset does not contain that wikipage in the first 1000 documents.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    "\n",
+    "# Print results\n",
+    "print(\"Query:\", query)\n",
+    "for doc_id in top_k.indices[0].tolist():\n",
+    "    print(docs[doc_id]['title'])\n",
+    "    print(docs[doc_id]['text'])\n",
+    "    print(docs[doc_id]['url'], \"\\n\")"
+   ]
+  },
+  {
```
Comment (arm-diaz) [inline cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:313]: Can we add a markdown section before this to explain why we are running a query on alan turing and another section after this code snippet with the conclusion? it should highlight the benefits of semantic search and keeping data up to date.

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+   "cell_type": "code",
```
Comment (arm-diaz) [inline cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:314]: Also, i'll leave this up to you as well as letting others comment in here. But I was expecting to see how the model could use the results of the semantic search to augment the response. Maybe, you can add a new section after the code snippet using the TOP K results from the semantic search as part of the prompt to then visualize how the model sticks to the provided data to get up to date results.

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 1 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:313]: Sure! Added a markdown section explaining that we are running the query on Alan Turing to get embeddings that capture the semantic meaning of the query. This is then used to the calculate the dot product which helps us determine the similarity between sentences.
Also added a conclusion as well as comments to explain that we loop over the indices of the top k relevant documents.

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 2 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:314]: Added a new section after the code snippet to explain how the TOP K results from the semantic search are useful when creating a prompt with context.

**Comment by HugoTse** [comment]
Added changes. Ready for new PR.

**Pull request closed without merging**